### PR TITLE
daemon, plugin: follow containerd namespace conventions

### PR DESCRIPTION
**Pull request by stevvooe**
Follow the conventions for namespace naming set out by other projects,
such as linuxkit and cri-containerd. Typically, they are some sort of
host name, with a subdomain describing functionality of the namespace.
In the case of linuxkit, services are launched in `services.linuxkit`.
In cri-containerd, pods are launched in `k8s.io`, making it clear that
these are from kubernetes.

Signed-off-by: Stephen J Day <stephen.day@docker.com>

This needs to be backported to the rc.

### CODE REVIEW

**Comment by stevvooe** [comment]
Failures are from `&errors.errorString{s:"Error response from daemon: \"/containerd-shim/containers.moby/40fae444a4b4ca63cdafe4f5e973c66c54e02490672fb11be2e02beb7928c97e/shim.sock\": unix socket path too long (limit 106): unknown"}`. I can fix that once we get a little feedback.

File: daemon/daemon.go
Comment (stevvooe) [inline daemon/daemon.go]: We can shorten this to just `moby`, and that will limit the impact.

File: plugin/executor/containerd/containerd.go
Comment (cpuguy83) [inline plugin/executor/containerd/containerd.go]: We'll need a way to migrate users from the old namespace. This is particularly a problem when live-restore is enabled... without live-restore we could just delete the old container and create a new one in the new namespace.

Even with live-restore we can probably get away with shutting down the containers and re-creating them in the new namespace.

Comment (thaJeztah) [reply plugin/executor/containerd/containerd.go]: Yes, we should have a look at migration, but impact is limited currently to people running 17.11 (which is an edge release); it's important we make sure that this change will be in 17.12.0 (worst case scenario: without the migration step, but instructions in the release notes to remove plugins and re-add them after updating)

File: daemon/daemon.go
Comment (vdemeester) [reply daemon/daemon.go]: Arf, I like `containers.moby` here though 😓.
If we need to do a migration anyway, let's do this one right (and thus I would like to have `containers.moby` instead).

File: plugin/executor/containerd/containerd.go
Comment (vdemeester) [reply plugin/executor/containerd/containerd.go]: @stevvooe @crosbymichael  is it possible to "change" an object namespace in containerd ?

**Review by seemethere — APPROVED** [review]
LGTM

File: plugin/executor/containerd/containerd.go
Comment (stevvooe) [reply plugin/executor/containerd/containerd.go]: The namespace cannot be renamed.

We should have considered this before merging containerd 1.0 support. We can migrate but the only impact here is that plugins will be impacted on upgrade.

Comment (seemethere) [reply plugin/executor/containerd/containerd.go]: As far as the impact goes we can include a note in the changelog with a link to a script that can do the migration.

I think that should suffice since the number of users using 17.11 along with plugins, in production, is probably low.

**Review by cpuguy83 — APPROVED** [review]
LGTM

File: plugin/executor/containerd/containerd.go
Comment (stevvooe) [reply plugin/executor/containerd/containerd.go]: I may have already answered this elsewhere, but no.

**Review by vieux — APPROVED** [review]
LGTM

**Pull request merged**
